jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -1382,6 +1384,42 @@ public void setSessionTrackingModes(Set<SessionTrackingMode> sessionTrackingMode
                 _sessionHandler.setSessionTrackingModes(sessionTrackingModes);
         }
 
+        @Override
+        public int getSessionTimeout()
+        {
+            if (!isStarting())
+                throw new IllegalStateException();
+            if (!_enabled)
+                throw new UnsupportedOperationException();
+
+            int timeout = -1;
+            if (_sessionHandler != null)
+            {
+                timeout = _sessionHandler.getMaxInactiveInterval();
+            }
+
+            return (int)TimeUnit.SECONDS.toMinutes(timeout);
```

Comment:
If there is no `SessionHandler` this returns `0`, and also if `SessionHandler` is configured with a value that is less than 60 seconds, either positive or negative.
If that's ok by the spec, I'm fine.

If you need to retain the fact that the value is negative, then perhaps this is enough?

```java
return timeout < 0 ? -1 : (int)TimeUnit.SECONDS.toMinutes(timeout);
```

Reply:
Any value of 0 or less has the same meaning as far as the spec is concerned, ie that the session never expires, so it's probably ok.

Reply:
Okay then.